src/Illuminate/Support/EncodedHtmlString.php

```diff
@@ -48,9 +59,23 @@ public static function convert($value, bool $withQuote = true, bool $doubleEncod
     #[\Override]
     public function toHtml()
     {
+        $value = $this->html;
+
+        if ($value instanceof DeferringDisplayableValue) {
+            $value = $value->resolveDisplayableValue();
+        }
+
+        if ($value instanceof Htmlable) {
+            return $value->toHtml();
+        }
+
+        if ($value instanceof BackedEnum) {
+            $value = $value->value;
+        }
```

Comment:
Copy the logic from `e()` function https://github.com/laravel/framework/blob/18a2f33c6c02e99bdb62bd2a0dc734fb189e29d2/src/Illuminate/Support/helpers.php#L127-L142